csharp/src/Apache.Arrow/Field.cs

```diff
@@ -45,5 +45,38 @@ public partial class Field
             IsNullable = nullable;
             Metadata = metadata?.ToDictionary(kv => kv.Key, kv => kv.Value);
         }
+
+        public override bool Equals(object obj)
+        {
+            if (this == obj)
+            {
+                return true;
+            }
+            if (obj == null || !this.GetType().Equals(obj.GetType()))
+            {
+                return false;
+            }
+
+            Field other = (Field)obj;
+            if (this.Name == other.Name && this.IsNullable == other.IsNullable &&
+                this.DataType.TypeId == other.DataType.TypeId)
+            {
+                if (this.HasMetadata && other.HasMetadata)
+                {
+                    return this.Metadata.Keys.Count() == other.Metadata.Keys.Count() &&
```

Comment:
Is there a more efficient way to do this? As it stands, this is 2 iterations over the dictionaries. While not really bad in this case, since presumably the metadata will usually be small, I wonder how this performs with huge dictionaries.